eZ/Publish/API/Repository/Tests/LocationServiceTest.php

```diff
@@ -3238,6 +3238,126 @@ public function testMoveSubtreeUpdatesPathIdentificationString(): void
+        $subFolderContent2 = $this->publishContentWithParentLocation('subFolderContent2', $sourceFolderContent->contentInfo->mainLocationId);
+        $targetFolderContent = $this->publishContentWithParentLocation('targetFolder', $mediaLocationId); // media/SourceFolder
+
+        $contentService->hideContent($subFolderContent1->contentInfo);
+
+        // Move media/SourceFolder to media/TargetFolder/
+        $locationService->moveSubtree(
+            $sourceFolderContent->contentInfo->getMainLocation(),
+            $targetFolderContent->contentInfo->getMainLocation()
+        );
+
+        $movedLocation = $locationService->loadLocation($sourceFolderContent->contentInfo->mainLocationId);
+        $newParentLocation = $locationService->loadLocation($targetFolderContent->contentInfo->mainLocationId);
+
+        // Assert Moved Location remains visible ( only child is hidden )
+        $this->assertPropertiesCorrect(
+            [
+                'hidden' => false,
+                'invisible' => false,
+                'depth' => $newParentLocation->depth + 1,
+                'parentLocationId' => $newParentLocation->id,
+                'pathString' => $newParentLocation->pathString . $this->parseId('location', $movedLocation->id) . '/',
+            ],
+            $movedLocation
+        );
+        self::assertFalse($movedLocation->getContentInfo()->isHidden);
+
+        // Assert children of Moved location
+        foreach ([$subFolderContent1->contentInfo->getMainLocation(), $subFolderContent2->contentInfo->getMainLocation()] as $childLocation) {
+            $this->assertPropertiesCorrect(
+                [
+                    'hidden' => $childLocation === $subFolderContent1->contentInfo->getMainLocation(), // Only SubFolderContent1 should be hidden,
+                    'invisible' => $childLocation === $subFolderContent1->contentInfo->getMainLocation(), // Only SubFolderContent1 should be hidden
+                    'depth' => $movedLocation->depth + 1,
+                    'parentLocationId' => $movedLocation->id,
+                    'pathString' => $movedLocation->pathString . $this->parseId('location', $childLocation->id) . '/',
+                ],
+                $childLocation
+            );
+            self::assertEquals($childLocation === $subFolderContent1->contentInfo->getMainLocation(), $childLocation->getContentInfo()->isHidden);
+        }
+    }
+
+    /**
+     * Test that is_visible is set correct for children when moving a content which is hidden (location is not hidden).
+     *
+     * @covers \eZ\Publish\API\Repository\LocationService::moveSubtree
+     * depends eZ\Publish\API\Repository\Tests\LocationServiceTest::testLoadLocation
+     */
+    public function testMoveSubtreeKeepsContentHiddenOnChildren(): void
+    {
+        $repository = $this->getRepository();
+
+        $mediaLocationId = $this->generateId('location', 43);
```

Comment:
Please create atomic test case instead of relying on existing legacy data with hardcoded database ids.

Reply:
@alongosz 
All other tests here uses the same approach, but ok... Instead of relying on existing Media folder with location `43`, I can just create a new folder. However, in order to do that I'll need to hardcode either `1` or `2` as parent. Is that OK?
The only alternative I found would be to do `LocationService::loadAllLocations(0,1)`, but is that any better ?

Reply:
> All other tests here uses the same approach, but ok... Instead of relying on existing Media folder with location `43`, I can just create a new folder. However, in order to do that I'll need to hardcode either `1` or `2` as parent. Is that OK? The only alternative I found would be to do `LocationService::loadAllLocations(0,1)`, but is that any better ?

It's a technical debt. We can hard-code for now Content root Location `2` because it's going to be easy to find & change once we switch to proper fixture data creation for tests. TBH that `2` probably should be a const for all tests, but that's out of scope here.

So indeed what we needed was
```php
       $sourceLocationId = $this->createFolder(
            [
                'eng-GB' => 'SourceParentFolder',
            ],
            2
        )->getVersionInfo()->getContentInfo()->mainLocationId;
```
I'm a bit late with the response, so to speed up the process I've allowed myself to fix this and small minor things I found and rebase the PR. Let's see if CS passes.